app/src/lib/stores/app-store.ts

```diff
@@ -664,6 +696,138 @@ export class AppStore extends TypedBaseStore<IAppState> {
     this.emitUpdate()
   }
 
+  private startAheadBehindUpdater(repository: Repository) {
+    if (this.currentAheadBehindUpdater) {
+      fatalError(
+        `An ahead/behind updater is already active and cannot start updating on ${
+          repository.name
+        }`
+      )
+
+      return
+    }
+
+    const updater = new AheadBehindUpdater(repository, aheadBehindCache => {
+      this.updateCompareState(repository, state => ({
+        aheadBehindCache,
+      }))
+      this.emitUpdate()
```

Comment:
I left out the `return` from this example because this function is a callback inside the `AheadBehindUpdater`:

https://github.com/desktop/desktop/blob/71e52ea0f3e48c12d127cd008f11d50a9351c27a/app/src/lib/stores/app-store.ts#L710-L715

The other examples use the latter pattern to mash together the right control flow with a void-returning function in more complex code. 

Happy to revisit this outside the PR and see if we can figure out how to be more consistent here.